src/components/Dashboard/SortList/SortList.jsx

```diff
@@ -0,0 +1,123 @@
+import React, { Component } from 'react';
+import {
+  Dropdown,
+  ButtonGroup,
+  Button,
+} from 'react-bootstrap';
+import storage from './storage';
+import './SortList.scss';
+import { ReactComponent as SortIcon } from './sort.svg';
+import { ReactComponent as SortIconAsc } from './sort-asc.svg';
+import { ReactComponent as SortIconDesc } from './sort-desc.svg';
+
+export default class SortList extends Component {
+  storageKey = 'null';
+
+  constructor(props) {
+    super(props);
+    const { storageKey: key } = this.props;
+    this.storageKey = key;
+
+    const stateDefaults = {
+      currentField: null,
+      currentDirection: 0,
+    };
+    const savedState = this.storageKey ? storage.get(this.storageKey) : {};
+    this.state = { ...stateDefaults, ...savedState };
+  }
+
+  onClickCallback = (field, direction) => {
+    this.setState({ currentField: field, currentDirection: direction });
+    if (this.storageKey) {
+      storage.set(this.storageKey, { currentField: field, currentDirection: direction });
+    }
+  };
+
+  render() {
+    const { fields, children, sortIconColor } = this.props;
+    const { currentField, currentDirection } = this.state;
+
+    const fieldsButtonGroup = fields.map((field) => {
+      const isHighlighted = field.key === currentField && currentDirection !== 0;
+      const isAsc = isHighlighted && currentDirection === 1;
+      const isDesc = isHighlighted && currentDirection === -1;
+
+      return (
+        <Dropdown.Item
+          className="btn-group btn-group-vertical dont-highlight p-0"
+          key={field.key}
+        >
+          <ButtonGroup
+            size="sm"
+            variant={isHighlighted ? 'light' : ''}
+          >
+            <Button
+              variant=""
+              className="rounded-0 text-left disabled btn-no-outline"
+            >
+              {field.text}
+            </Button>
+            <Button
+              variant="light"
+              active={isAsc}
+              className="rounded-0 btn-no-outline flex-grow-0"
+              onClick={() => this.onClickCallback(field.key, 1)}
+            >
+              <SortIconAsc />
+            </Button>
+            <Button
+              variant="light"
+              active={isDesc}
+              className="rounded-0 btn-no-outline flex-grow-0"
+              onClick={() => this.onClickCallback(field.key, -1)}
+            >
+              <SortIconDesc />
+            </Button>
+          </ButtonGroup>
+        </Dropdown.Item>
+      );
+    });
+
+    // Copied children array for immutability.
+    const sortedChildrenList = [...children].sort((a, b) => {
```

Comment:
The sorting function is pretty massive. Why not move it to a separate file with all other auxiliary functions?

Reply:
Sort is mutable funtion, so it sorts source array. And if we want to cancel sorting direction, e won't be able to do it.

https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/sort